src/corecel/sys/KernelParamCalculator.device.hh

```diff
@@ -128,6 +129,8 @@ class KernelParamCalculator
 
     void register_kernel(std::string_view name, KernelAttributes&& attributes);
     void log_launch(size_type min_num_threads) const;
+    template<class F>
+    void set_carveout(F* kernel_func_ptr) const;
```

Comment:
setting the carveout ratio here works and requires minimal changes but I'm not convinced it makes sense semantically since the class is called `KernelParamCalculator` 🤔

Reply:
Well, shared memory is a kernel launch parameter so maybe it makes sense?